docs/concepts/types.md

```diff
@@ -460,10 +460,12 @@ class Model(BaseModel):
     name: Username
 
 
-assert Model(name='ABC').name == 'abc'
+assert Model(name='ABC').name == 'abc'  # (2)!
 ```
 
-Notice that type checkers will not complain about assigning `'abc'` to `Username` like they did in the previous example because they do not consider `Username` to be a distinct type from `str`.
+1. The `frozen=True` specification is not strictly necessary, but renders `MyAfterValidator` as immutable, which triggers an addition of the `__hash__`
+method to the `dataclass`, which is sometimes used during serialization.
+2. Notice that type checkers will not complain about assigning `'abc'` to `Username` like they did in the previous example because they do not consider `Username` to be a distinct type from `str`.
```

Comment:
```suggestion
2. Notice that type checkers will not complain about assigning `'ABC'` to `Username` like they did in the previous example because they do not consider `Username` to be a distinct type from `str`.
```